tidy3d/plugins/resonance/resonance.py

```diff
@@ -0,0 +1,272 @@
+"""Find resonances in time series data
+"""
+
+from typing import Tuple
+from functools import partial
+
+import numpy as np
+import scipy.linalg
+import xarray as xr
+
+from pydantic import Field
+
+from ...components.base import Tidy3dBaseModel
+from ...components.data.data_array import ScalarFieldTimeDataArray
+from ...constants import HERTZ
+from ...log import SetupError
+
+# ResonanceData will be used internally
+class ResonanceData(Tidy3dBaseModel):
+    """Data class for storing objects computed while running the solver."""
+
+    eigvals: np.ndarray[complex] = Field(
+        ..., title="Eigenvalues", description="Resonance eigenvalues."
+    )
+    complex_amplitudes: np.ndarray[complex] = Field(
+        None, title="Complex amplitudes", description="Complex resonance amplitudes"
+    )
+    errors: np.ndarray[float] = Field(
+        None, title="Errors", description="Rough eigenvalue error estimate."
+    )
+
+
+class ResonanceFinder(Tidy3dBaseModel):
+    """Tool that extracts resonance information (frequency, decay rate,
+    amplitude, and phase) from a time series of the following form.
+
+    Note
+    ----
+    .. math::
+
+        f(t) = \\sum_k a_k e^{i \\phi_k} e^{-2 \\pi i f_k t - \\lambda_k t}
+
+    Note
+    ----
+
+    We use the algorithm described in
+
+    Vladimir A. Mandelshtam and Howard S. Taylor,
+    "Harmonic inversion of time signals and its applications,"
+    J. Chem. Phys. 107, 6756 (1997).
+
+    Example
+    -------
+    from tidy3d.plugins import ResonanceFinder
+    t = np.linspace(0, 10000, 10000)
+    f1 = 0.1 * 2 * np.pi - 0.01j
+    f2 = 0.08 * 2 * np.pi - 0.001j
+    data = np.exp(-f1 * t * 1j) - 2 * np.exp(-f2 * t * 1j)
+    resonances = ResonanceFinder(freq_window=(0.05,0.15)).run(data, 1)
+    """
+
+    freq_window: Tuple[float, float] = Field(
```

Comment:
what happens if `freq_window[1] < freq_window[0]`? we might need a validator here.